Python/formatter_unicode.c

```diff
@@ -1151,18 +1240,20 @@ format_float_internal(PyObject *value,
 
     /* Determine if we have any "remainder" (after the digits, might include
        decimal or exponent or both (or neither)) */
-    parse_number(unicode_tmp, index, index + n_digits, &n_remainder, &has_decimal);
+    parse_number(unicode_tmp, index, index + n_digits,
+                 &n_remainder, &n_frac, &has_decimal);
 
     /* Determine the grouping, separator, and decimal point, if any. */
     if (get_locale_info(format->type == 'n' ? LT_CURRENT_LOCALE :
                         format->thousands_separators,
+                        format->frac_thousands_separator,
```

Comment:
What do with the locale? At the very least I would reject specifying the separator manually:

```python
>>> import locale
>>> locale.setlocale(locale.LC_ALL, 'af_za')
>>> print(f"{123_456.123_456:.12,n}")
123.456,123,456 # <- 2 decimal points
```

Would it be a breaking change to do this?

```python
>>> print(f"{123_456.123_456:.12n}")
123.456,123.456
```

Reply:
I think we should just reject separators with the ``n`` type.

Reply:
It's not possible to use the underscore separator with `n` format:

```
>>> format(1234, "_n")
ValueError: Cannot specify '_' with 'n'.
```

So I agree that we should reject separators with the `n` type.

Reply:
See invalid_thousands_separator_type() in parse_internal_render_format_spec().

Reply:
@vstinner, this check already added.  Unfortunately, that means with ``n`` type we can have locale-dependent separators only for integer part.

Reply:
BTW, should we allow *different* separators?  So far it's permitted:
```pycon
>>> f"{122222.22222:_.7,f}"
'122_222.222,220,0'
```